File: problems/1431/1431.kids-with-the-greatest-number-of-candies.py
#
# @lc app=leetcode id=1431 lang=python3
#
# [1431] Kids With the Greatest Number of Candies
#


# @lc code=start
class Solution:
    def kidsWithCandies(
        self, candies: List[int], extraCandies: int
    ) -> List[bool]:
        n = len(candies)
        curr_max = max(candies)
        res = [False] * n

        for i, c in enumerate(candies):
            if c + extraCandies >= curr_max:
                res[i] = True

        return res


# @lc code=end

File: problems/1431/1431.kids-with-the-greatest-number-of-candies.go
/*
 * @lc app=leetcode id=1431 lang=golang
 *
 * [1431] Kids With the Greatest Number of Candies
 */

package solution

// @lc code=start
func kidsWithCandies(candies []int, extraCandies int) []bool {
	n := len(candies)
	currMax := max(candies)
	res := make([]bool, n)

	for i, val := range candies {
		if val+extraCandies >= currMax {
			res[i] = true
		}
	}

	return res
}

func max(arr []int) int {
	currMax := 0
	for _, val := range arr {
		if val > currMax {
			currMax = val
		}
	}

	return currMax
}

// @lc code=end
